src/tests/unit/utils/PersonIdentityUtils.test.ts

```diff
@@ -1,33 +1,43 @@
-import { getNameByType } from "../../../utils/PersonIdentityUtils";
+import { getFirstName, getMiddleNames, getLastName, getFullName } from "../../../utils/PersonIdentityUtils";
 
 describe("PersonIdentityUtils", () => {
-	describe("#getNameByType", () => {
-		it("returns all full name if no name type provided", () => {
-			const result = getNameByType([{ nameParts: [
-				{ type: "GivenName", value: "First" },
-				{ type: "GivenName", value: "Middle" },
-				{ type: "FamilyName", value: "Last" },
-			] }]);
-			expect(result).toBe("First Middle Last");
-		});
 
-		it("returns name based on type - GivenName", () => {
-			const result = getNameByType([{ nameParts: [
-				{ type: "GivenName", value: "First" },
-				{ type: "GivenName", value: "Middle" },
-				{ type: "FamilyName", value: "Last" },
-			] }], "GivenName");
-			expect(result).toBe("First Middle");
-		});
+	it("returns all full name", () => {
+		const result = getFullName([{ nameParts: [
+			{ type: "GivenName", value: "First" },
+			{ type: "GivenName", value: "Middle" },
+			{ type: "FamilyName", value: "Last" },
+		] }]);
+		expect(result).toBe("First Middle Last");
+	});
 
-		it("returns name based on type - FamilyName", () => {
-			const result = getNameByType([{ nameParts: [
-				{ type: "GivenName", value: "First" },
-				{ type: "GivenName", value: "Middle" },
-				{ type: "FamilyName", value: "Last" },
-			] }], "FamilyName");
-			expect(result).toBe("Last");
-		});
+	it("returns first name", () => {
+		const result = getFirstName([{ nameParts: [
+			{ type: "GivenName", value: "First" },
+			{ type: "GivenName", value: "Middle" },
+			{ type: "FamilyName", value: "Last" },
+		] }]);
+		expect(result).toBe("First");
+	});
 
+	it("returns middle names", () => {
+		const result = getMiddleNames([{ nameParts: [
+			{ type: "GivenName", value: "First" },
+			{ type: "GivenName", value: "MiddleOne" },
+			{ type: "GivenName", value: "MiddleTwo" },
+			{ type: "FamilyName", value: "Last" },
+		] }]);
+		expect(result).toBe("MiddleOne MiddleTwo");
 	});
+
+	it("returns last name", () => {
```

Comment:
Can we add a test for no first name and empty name to make the sure the code performs correctly

Reply:
Also what happens if we call getMiddleNames with just one middle name

Reply:
First name / family names are required fields in CIC, is it realistic that we would ever receive these as empty values?

`getMiddleNames` will respond with a string containing single or multiple values depending on the payload, and an empty string if none are provided